File: trunk/EventLib/tests/static/javascript/TestStatus.js
// $Id$
//
// Test status object
//

// Create test suite object
function TestStatus() {
}

// Specify order of tests
TestStatus.exposeTestFunctionNames = function() {
    return [ "testStatusCreate1"
           , "testStatusCreate2"
           , "testStatusCreate3"
           , "testStatusEqual1"
           , "testStatusEqual2"
           , "testStatusEqual3"
           , "testStatusString1"
           , "testStatusString2"
           , "testStatusString3"
           , "testStatusString4"
           , "testStatusException1"
           , "testStatusException2"
           , "testStatusEvent1"
           , "testStatusEvent2"
           , "testStatusValues"
           , "testXXX"
           ];
}

// Setup and teardown

TestStatus.prototype.setUp = function() {
}

TestStatus.prototype.tearDown = function() {
}

// Test cases

TestStatus.prototype.testStatusCreate1 = function() {
    var s = new Status("StatusUri", "message", "values");
    assertEquals("testStatusCreate1, getUri",     "StatusUri", s.getUri());
    assertEquals("testStatusCreate1, getMessage", "message",   s.getMessage());
    assertEquals("testStatusCreate1, getValues",  "values",    s.getValues());
}

TestStatus.prototype.testStatusCreate2 = function() {
    var s = new Status("StatusUri", "message");
    assertEquals("testStatusCreate2, getUri",     "StatusUri", s.getUri());
    assertEquals("testStatusCreate2, getMessage", "message",   s.getMessage());
    assertEquals("testStatusCreate2, getValues",  null,        s.getValues());
}

TestStatus.prototype.testStatusCreate3 = function() {
    var s = new Status("StatusUri");
    assertEquals("testStatusCreate3, getUri",     "StatusUri", s.getUri());
    assertEquals("testStatusCreate3, getMessage", "", s.getMessage());
    assertEquals("testStatusCreate3, getValues",  null, s.getValues());
}

TestStatus.prototype.testStatusEqual1 = function() {
    var s1 = new Status("StatusUri", "message1", "values1");
    var s2 = new Status("StatusUri", "message2", "values2");
    assertEquals("testStatusEqual1", s1, s2);
}

TestStatus.prototype.testStatusEqual2 = function() {
    var s1 = new Status("StatusUri");
    var s2 = new Status("StatusUri");
    assertEquals("testStatusEqual2", s1, s2);
}

TestStatus.prototype.testStatusEqual3 = function() {
    var s1 = new Status("StatusUri1", "message", "values");
    var s2 = new Status("StatusUri2", "message", "values");
    assertNotEquals("testStatusEqual3", s1, s2);
}

TestStatus.prototype.testStatusString1 = function() {
    var s = new Status("StatusUri", "message", "values");
    assertEquals("testStatusString1", "StatusUri: message", s.toString());
}

TestStatus.prototype.testStatusString2 = function() {
    var s = new Status("StatusUri", "message");
    assertEquals( "testStatusString2", "StatusUri: message", s.toString() );
}

TestStatus.prototype.testStatusString3 = function() {
    var s = new Status("StatusUri");
    assertEquals( "testStatusString3", "StatusUri", s.toString() );
}

TestStatus.prototype.testStatusString4 = function() {
    var s = new Status("StatusUri", null);
    assertEquals( "testStatusString4", "StatusUri", s.toString() );
}

TestStatus.prototype.testStatusException1 = function() {
    var s = new Status("StatusUri", "message", "values");
    try {
        throw s.makeException();
    }
    catch (e) {
        assertEquals( "testStatusException1, str",       "StatusUri: message", e.toString());
        assertEquals( "testStatusException1, getStatus", s,                    e.getStatus());
    }
}

TestStatus.prototype.testStatusException2 = function() {
    var s = new Status("StatusUri");
    try {
        throw s.makeException();
    }
    catch (e) {
        assertEquals( "testStatusException2, toString",  "StatusUri", e.toString() );
        assertEquals( "testStatusException2, getStatus", s,           e.getStatus() );
    }
}

TestStatus.prototype.testStatusEvent1 = function() {
    var s = new Status("StatusUri", "message", "values");
    var e = s.makeEvent("SourceUri");
    var ee = new Event("StatusUri", "SourceUri", s);
    assertEquals("testStatusEvent1", ee, e);
}

TestStatus.prototype.testStatusEvent2 = function() {
    var s = new Status("StatusUri");
    var e = s.makeEvent("SourceUri");
    var ee = new Event("StatusUri", "SourceUri", s);
    assertEquals("testStatusEvent2", ee, e);
}

TestStatus.prototype.testStatusValues = function() {
    var u = StatusUri.OK;
    var m = "";
    var v = null;
    assertEquals("testStatusValues, StatusVal.OK.getUri",     u, StatusVal.OK.getUri());
    assertEquals("testStatusValues, StatusVal.OK.getMessage", m, StatusVal.OK.getMessage());
    assertEquals("testStatusValues, StatusVal.OK.getValues",  v, StatusVal.OK.getValues());
}

TestStatus.prototype.testXXX = function() {
    assertEquals( "testXXX", 4, 2+2 );
}

// Expose functions at global level of frame for JSUnit

// Specify order of tests
exposeTestFunctionNames = TestStatus.exposeTestFunctionNames;

// JSUnit page setup
function setUpPage() {
    setUpPageForJSUnit(TestStatus, this);
}

// End.

File: EventLib/tests/TestStatus.py
# $Id$
#
# Copyright (c) 2008 WebBrick Systems Limited
# Released under the MIT licence
# See LICENCE.TXT included with these files,
# or http://www.opensource.org/licenses/mit-license.php
#
# Unit test for status value type and handling
# See http://pyunit.sourceforge.net/pyunit.html
#

import sys
import unittest

sys.path.append("../..")

from EventLib.Status import Status, StatusVal, StatusException
from EventLib.Event  import Event

# Test class
class TestStatusVals(unittest.TestCase):

    def setUp(self):
        return

    def tearDown(self):
        return

    # Test cases

    def testStatusCreate1(self):
        s = Status("StatusUri", "message", "values")
        assert s.getUri()     == "StatusUri", "testStatusCreate1, getUri"
        assert s.getMessage() == "message",   "testStatusCreate1, getMessage"
        assert s.getValues()  == "values",    "testStatusCreate1, getValues"

    def testStatusCreate2(self):
        s = Status("StatusUri", values="values", message="message")
        assert s.getUri()     == "StatusUri", "testStatusCreate2, getUri"
        assert s.getMessage() == "message",   "testStatusCreate2, getMessage"
        assert s.getValues()  == "values",    "testStatusCreate2, getValues"

    def testStatusCreate3(self):
        s = Status("StatusUri")
        assert s.getUri()     == "StatusUri", "testStatusCreate3, getUri"
        assert s.getMessage() == "",          "testStatusCreate3, getMessage"
        assert s.getValues()  == None,        "testStatusCreate3, getValues"

    def testStatusEqual1(self):
        s1 = Status("StatusUri", "message1", "values1")
        s2 = Status("StatusUri", "message2", "values2")
        assert s1 == s2, "testStatusEqual1"

    def testStatusEqual2(self):
        s1 = Status("StatusUri")
        s2 = Status("StatusUri")
        assert s1 == s2, "testStatusEqual2"

    def testStatusEqual3(self):
        s1 = Status("StatusUri1", "message", "values")
        s2 = Status("StatusUri2", "message", "values")
        assert s1 != s2, "testStatusEqual3"

    def testStatusString1(self):
        s = Status("StatusUri", "message", "values")
        assert str(s) == "StatusUri: message", "testStatusString1"

    def testStatusString2(self):
        s = Status("StatusUri", values="values", message="message")
        assert str(s) == "StatusUri: message", "testStatusString2"

    def testStatusString3(self):
        s = Status("StatusUri")
        assert str(s) == "StatusUri", "testStatusString3"

    def testStatusString4(self):
        s = Status("StatusUri", message=None)
        assert str(s) == "StatusUri", "testStatusString4"

    def testStatusException1(self):
        s = Status("StatusUri", "message", "values")
        try:
            raise s.makeException()
        except StatusException, e:
            assert str(e) == "StatusUri: message", "testStatusException1, str"
            assert e.getStatus() == s, "testStatusException1, getStatus"

    def testStatusException2(self):
        s = Status("StatusUri")
        try:
            raise s.makeException()
        except StatusException, e:
            assert str(e) == "StatusUri", "testStatusException2, str"
            assert e.getStatus() == s, "testStatusException2, getStatus"

    def testStatusEvent1(self):
        s = Status("StatusUri", "message", "values")
        e = s.makeEvent("SourceUri")
        assert e == Event("StatusUri", "SourceUri", s), "testStatusEvent1"

    def testStatusEvent2(self):
        s = Status("StatusUri")
        e = s.makeEvent("SourceUri")
        assert e == Event("StatusUri", "SourceUri", s), "testStatusEvent2"

    def testStatusValues(self):
        assert isinstance(StatusVal.OK, Status), "testStatusValues, StatusVal.OK"

    # Sentinel/placeholder tests

    def testUnits(self):
        assert (True)

    def testComponents(self):
        assert (True)

    def testIntegration(self):
        assert (True)

# Assemble test suite
from MiscLib import TestUtils

def getTestSuite(select="unit"):
    """
    Get test suite

    select  is one of the following:
            "unit"      return suite of unit tests only
            "component" return suite of unit and component tests
            "all"       return suite of unit, component and integration tests
            "pending"   return suite of pending tests
            name        a single named test to be run
    """
    testdict = {
        "unit": 
            [ "testUnits"
            , "testStatusCreate1"
            , "testStatusCreate2"
            , "testStatusCreate3"
            , "testStatusEqual1"
            , "testStatusEqual2"
            , "testStatusEqual3"
            , "testStatusString1"
            , "testStatusString2"
            , "testStatusString3"
            , "testStatusString4"
            , "testStatusException1"
            , "testStatusException2"
            , "testStatusEvent1"
            , "testStatusEvent2"
            , "testStatusValues"
            ],
        "component":
            [ "testComponents"
            ],
        "integration":
            [ "testIntegration"
            ],
        "pending":
            [ "testPending"
            ]
        }
    return TestUtils.getTestSuite(TestStatusVals, testdict, select=select)

# Run unit tests directly from command line
if __name__ == "__main__":
    TestUtils.runTests("TestStatus.log", getTestSuite, sys.argv)

# End.
